Gems/AtomLyIntegration/CommonFeatures/Code/Source/Decals/EditorDecalComponent.cpp

```diff
@@ -97,13 +97,32 @@ namespace AZ
             BaseClass::Deactivate();
         }
 
-        AZ::Transform EditorDecalComponent::GetTransform() const
+        AZ::Transform EditorDecalComponent::GetWorldTransform() const
         {
             AZ::Transform transform = AZ::Transform::CreateIdentity();
             AZ::TransformBus::EventResult(transform, GetEntityId(), &AZ::TransformBus::Events::GetWorldTM);
             return transform;
         }
 
+        AZ::Vector3 EditorDecalComponent::GetNonUniformScale() const
+        {
+            AZ::Vector3 nonUniformScale;
+            AZ::NonUniformScaleRequestBus::EventResult(nonUniformScale, GetEntityId(), &AZ::NonUniformScaleRequests::GetScale);
+            return nonUniformScale;
+        }
+
+        AZ::Matrix3x4 EditorDecalComponent::GetWorldTransformWithNonUniformScale() const
+        {
+            const AZ::Transform worldTransform = GetWorldTransform();
+            const AZ::Matrix3x3 rotationMat = AZ::Matrix3x3::CreateFromQuaternion(worldTransform.GetRotation());
+
+            const AZ::Vector3 nonUniformScale = GetNonUniformScale() * worldTransform.GetUniformScale();
+            const AZ::Matrix3x3 nonUniformScaleMat = AZ::Matrix3x3::CreateScale(nonUniformScale);
+            const AZ::Matrix3x3 rotationAndScale = rotationMat * nonUniformScaleMat;
+
+            return AZ::Matrix3x4::CreateFromMatrix3x3AndTranslation(rotationAndScale, worldTransform.GetTranslation());
```

Comment:
Can you cache the transform matrix and only update it when the transform component is changes or the nonUniformScale component is changed?

Reply:
I gave this a shot, unfortunately I was unable to cache because the base class of this thing was already using the RegisterScaleChangedEvent() and it doesn't seem like I can register it twice from parent and child.

While I was debugging this I *did* notice that the base class was caching the non-uniform scale. I just grabbed that and replaced 10 lines of code with just using that protected variable..